File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Controllers/TableViewControllers/TableViewCells/BtTextFieldTableViewCell.cs
using System;

using Foundation;
using UIKit;

namespace bugTrapKit
{
	public partial class BtTextFieldTableViewCell : BtNewBugDetailsTableViewCell, IUITextFieldDelegate
	{

		public UITextField TxtField {
			get { return TextField; }
		}

		public BtTextFieldTableViewCell (IntPtr handle) : base(handle)
		{
			
		}


		public override void AwakeFromNib ()
		{
			base.AwakeFromNib();

			PlaceholderLabel.Hidden = TextField.HasText;
		}


		public override void SetData (CellData cellData)
		{
			var detail = cellData.Get(DataKeys.Detail);
			var auxiliary = cellData.Get(DataKeys.Auxiliary);

			TextField.Text = detail;
			PlaceholderLabel.Hidden = TextField.HasText;
			PlaceholderLabel.Text = string.IsNullOrEmpty(detail) ? auxiliary : null;
		}


		public bool TextFieldShouldBecomeFirstResponder ()
		{
			var respond = !TextField.HasText;

			if (respond) TextField.BecomeFirstResponder();

			return respond;
		}


		[Export("textField:shouldChangeCharactersInRange:replacementString:")]
		public bool ShouldChangeCharacters (UITextField textField, NSRange range, string replacementString)
		{
			PlaceholderLabel.Hidden = !string.IsNullOrEmpty(replacementString) || range.Location > 0;
			return true;
		}


		[Export("textFieldShouldReturn:")]
		public bool ShouldReturn (UITextField textField)
		{
			textField.ResignFirstResponder();
			return false;
		}
	}
}

File: iOS/Code/Swift/bugTrap/bugTrapKit/Controllers/TableViewControllers/TableViewCells/BtTextFieldTableViewCell.swift
//
//  BtTextFieldTableViewCell.swift
//  bugTrap
//
//  Created by Colby L Williams on 6/30/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import UIKit

class BtTextFieldTableViewCell: BtNewBugDetailsTableViewCell, UITextFieldDelegate {
	
	@IBOutlet weak var TextField: UITextField!
	@IBOutlet weak var PlaceholderLabel: UILabel!
	
	override func awakeFromNib() {
		super.awakeFromNib()
		
		PlaceholderLabel.hidden	= TextField.hasText()
	}
	
	
	func textField(textField: UITextField, shouldChangeCharactersInRange range: NSRange, replacementString string: String) -> Bool {
		PlaceholderLabel.hidden	= !string.isEmpty || range.location > 0
		return true
	}

	
	override func setData(title: String? = nil, detail: String? = nil, placeholder: String? = nil, auxiliary: String? = nil) {
		TextField.text = detail ?? ""
		PlaceholderLabel.hidden	= TextField.hasText()
		PlaceholderLabel.text = detail == nil || detail!.isEmpty ? auxiliary ?? "" : ""
	}
	
	
	func textFieldShouldBecomeFirstResponder() -> Bool {
		if !TextField.hasText() {
			TextField.becomeFirstResponder()
			return true
		}
		return false
	}
	
	func textFieldShouldReturn(textField: UITextField) -> Bool {
		TextField.resignFirstResponder()
		return false
	}
}